sno/init.py

```diff
@@ -503,6 +520,84 @@ def build_meta_info(self):
         yield from gpkg.get_meta_info(db, layer=self.table)
 
 
+class ImportPostgreSQL(OgrImporter):
+    @classmethod
+    def postgres_url_to_ogr_conn_str(cls, url):
+        """
+        Takes a URL ('postgresql://..')
+        and turns it into a key/value connection string, prefixed by 'PG:' for OGR.
+
+        libpq actually handles URIs fine, but OGR doesn't :(
+        So to import via OGR we have to convert them.
+
+        https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING
+
+        ^ These docs say these URLs can contain multiple hostnames or ports,
+        but we don't handle that.
+        """
+
+        url = urlsplit(url)
+        scheme = url.scheme.lower()
+        if scheme not in ('postgres', 'postgresql'):
+            raise ValueError("Bad scheme")
+
+        params = dict(parse_qsl(url.query))
+
+        if url.username:
+            params.setdefault('user', url.username)
```

Comment:
Looks like "user" could be specified in two different places, just like the other fields, and one has precedence. Add a comment or add something about it to the docstring.